### PR TITLE
Remove unused forward_worker module

### PR DESCRIPTION
**Pull request by KirillLykov**
#### Problem

Remove unused module. Discussion starts here https://discord.com/channels/428295358100013066/439194979856809985/1293597233400512593

### CODE REVIEW

File: core/src/banking_stage.rs
Comment (apfitzge) [inline core/src/banking_stage.rs:1]: line 77; we should remove the dead_code from scheduler_messages, and in `scheduler_messages` we should remove the `ForwardWork` message types

Comment (KirillLykov) [reply core/src/banking_stage.rs:1]: addressed this one

**Comment by KirillLykov** [comment]
failing CI due to unrelated test:

```
failures:
--
  |  
  | ---- wen_restart::tests::test_wen_restart_normal_flow stdout ----
  | thread 'solana-wen-restart' panicked at wen-restart/src/wen_restart.rs:1649:17:
  | assertion failed: wait_for_wen_restart(wen_restart_config).is_ok()
  | thread 'wen_restart::tests::test_wen_restart_normal_flow' panicked at wen-restart/src/wen_restart.rs:1752:9:
  | assertion failed: wen_restart_thread_handle.join().is_ok()

```

**Review by apfitzge — APPROVED** [review]
Thanks for removing this dead code!

**Pull request merged**